### PR TITLE
Migrate to new spine

### PR DESCRIPTION
**Pull request by dmdashenkov**
### What is made:
- Migrated to new version of the Spine framework.
- Updated protobuf gradle plugin version and third party dependencies.
- Fixed all tests.
### To review:
- For testing needs aggregates now have extra command handler methods.
- Some proto definitions that are required for testing are now located in `main` folder and are used in prod code.
- `LobbyAggregate` as a superclass for all the aggregates of the project was introduced mainly for testing needs too.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol PTAL

File: common/src/main/java/org/spine3/samples/lobby/common/util/aggregate/LobbyAggregate.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import org.spine3.samples.lobby.common.ImportEvents;
+import org.spine3.server.aggregate.Aggregate;
+
+import java.util.List;
+
+/**
+ * @author Dmytro Dashenkov
+ */
+@SuppressWarnings("AbstractClassWithoutAbstractMethods")
+public abstract class LobbyAggregate<I, S extends Message, B extends Message.Builder> extends Aggregate<I, S, B> {
```
Comment (armiol) [inline common/src/main/java/org/spine3/samples/lobby/common/util/aggregate/LobbyAggregate.java]: Please add class-level JavaDocs. This is a requirement for all classes in Spine.

File: common/src/main/proto/spine/test/aggregate/commands.proto [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+option (type_url_prefix) = "type.spine3.org";
+option java_package="org.spine3.samples.lobby.common";
+option java_outer_classname = "AggregateCommandsProto";
+option java_multiple_files = true;
+
+import "spine/annotations.proto";
+import "spine/test/aggregate/project.proto";
+import "spine/base/event.proto";
+
+message CreateProject {
```
Comment (armiol) [inline common/src/main/proto/spine/test/aggregate/commands.proto]: Please describe each of new messages via `// ...comments`. See `spine-core` for examples and styling.

File: registration/src/main/java/org/spine3/samples/lobby/registration/conference/ConferenceProjection.java [hunk 3]
```diff
@@ -156,27 +160,57 @@ private void addOrRemoveSeats(int newQuantity, int oldQuantity, SeatTypeId id) {
...
-                .build();
+                                         .setConferenceId(getState().getId())
+                                         .setQuantity(newSeatQuantity(seatTypeId, quantity))
+                                         .build();
         final Command command = create(message, newCommandContext());
-        commandBus.post(command);
+        commandBus.post(command, new StreamObserver<Response>() {
+            @Override
+            public void onNext(Response value) {
+
```
Comment (armiol) [inline registration/src/main/java/org/spine3/samples/lobby/registration/conference/ConferenceProjection.java]: Describe why these methods are empty. Use comments.

File: registration/src/main/java/org/spine3/samples/lobby/registration/order/OrderAggregate.java [hunk 4]
```diff
@@ -154,22 +157,29 @@ public OrderConfirmed handle(ConfirmOrder command, CommandContext context) {
...
+                                                                      .setOrderId(command.getOrderId())
+                                                                      .setPersonalInfo(command.getRegistrant())
+                                                                      .build();
         return result;
     }
 
+    @VisibleForTesting
+    @Assign
+    @Override
+    public List<Event> handle(ImportEvents command, CommandContext ctx) {
```
Comment (armiol) [inline registration/src/main/java/org/spine3/samples/lobby/registration/order/OrderAggregate.java:178]: Do you really need `public`?

File: common/src/main/java/org/spine3/samples/lobby/common/util/aggregate/LobbyAggregate.java (hunk 1 above)
Comment (armiol) [inline common/src/main/java/org/spine3/samples/lobby/common/util/aggregate/LobbyAggregate.java]: Let's rename it to `AbstractLobbyAggregate`.

File: registration/src/main/java/org/spine3/samples/lobby/registration/order/OrderAggregate.java (hunk 4 above)
Comment (dmdashenkov) [reply registration/src/main/java/org/spine3/samples/lobby/registration/order/OrderAggregate.java:178]: Can we have non-public command handlers?

File: registration/src/test/java/org/spine3/samples/lobby/registration/util/AnyExtracterUtil.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+import com.google.protobuf.Any;
+import com.google.protobuf.Message;
+import org.spine3.protobuf.AnyPacker;
+
+/**
+ * @author Dmytro Dashenkov
+ */
+@SuppressWarnings("UtilityClass")
+public class AnyExtracterUtil {
```
Comment (armiol) [inline registration/src/test/java/org/spine3/samples/lobby/registration/util/AnyExtracterUtil.java]: Documentation is required.

Comment (armiol) [inline registration/src/test/java/org/spine3/samples/lobby/registration/util/AnyExtracterUtil.java]: Also, this is `MessagePacker`. Try avoiding `...Util` suffix.

File: registration/src/test/java/org/spine3/samples/lobby/registration/util/ImportCommandUtil.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import org.spine3.base.EventContext;
+import org.spine3.base.Events;
+import org.spine3.samples.lobby.common.ImportEvents;
+import org.spine3.samples.lobby.common.util.aggregate.LobbyAggregate;
+
+/**
+ * @author Dmytro Dashenkov
+ */
+@SuppressWarnings("UtilityClass")
+public class ImportCommandUtil {
```
Comment (armiol) [inline registration/src/test/java/org/spine3/samples/lobby/registration/util/ImportCommandUtil.java]: Documentation is required.
Also, this is `CommandImporter`.

File: webapp/web/index.jsp [hunk 7]
```diff
@@ -1,9 +1,9 @@
 <%@ page contentType="text/html;charset=UTF-8" language="java" %>
```
Comment (armiol) [inline webapp/web/index.jsp:1]: Why change this file?

File: webapp/web/WEB-INF/appengine-web.xml [hunk 8]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <appengine-web-app xmlns="http://appengine.google.com/ns/1.0">
```
Comment (armiol) [inline webapp/web/WEB-INF/appengine-web.xml:2]: Why change this file?

File: build.gradle [hunk 9]
```diff
@@ -66,8 +70,8 @@ subprojects {
         compile group: 'com.google.code.findbugs', name: 'jsr305', version: '3.0.0'
 
         // SLF4J
-        compile group: 'org.slf4j', name: 'slf4j-api', version: '1.7.12'
-        compile group: 'org.slf4j', name: 'slf4j-jdk14', version: '1.7.12'
+        compile group: 'org.slf4j', name: 'slf4j-api', version: '1.7.21'
```
Comment (armiol) [inline build.gradle]: Please extract the version into a constant.

**Comment by armiol** [comment]
@dmdashenkov please see my comments.

File: webapp/web/WEB-INF/appengine-web.xml (hunk 8 above)
Comment (dmdashenkov) [reply webapp/web/WEB-INF/appengine-web.xml:2]: Seems that Gradle changed it. Same applies to the `webapp/web/index.jsp` file.

**Comment by dmdashenkov** [comment]
@armiol PTAL

File: build.gradle [hunk 10]
```diff
@@ -70,8 +71,8 @@ subprojects {
         compile group: 'com.google.code.findbugs', name: 'jsr305', version: '3.0.0'
 
         // SLF4J
-        compile group: 'org.slf4j', name: 'slf4j-api', version: '1.7.21'
-        compile group: 'org.slf4j', name: 'slf4j-jdk14', version: '1.7.21'
+        compile group: 'org.slf4j', name: 'slf4j-api', version: SLF4J_VERSION
+        compile group: 'org.slf4j', name: 'slf4j-jdk14', version: project.SLF4J_VERSION
```
Comment (armiol) [inline build.gradle:75]: Why the notation is different? I.e. `SLF4J_VERSION` vs. `project.SLF4J_VERSION`?

File: common/src/main/java/org/spine3/samples/lobby/common/util/aggregate/AbstractLobbyAggregate.java [hunk 11]
```diff
@@ -30,26 +30,40 @@
...
  */
 @SuppressWarnings("AbstractClassWithoutAbstractMethods")
-public abstract class LobbyAggregate<I, S extends Message, B extends Message.Builder> extends Aggregate<I, S, B> {
+public abstract class AbstractLobbyAggregate<I, S extends Message, B extends Message.Builder> extends Aggregate<I, S, B> {
 
     /**
      * Creates a new aggregate instance.
      *
      * @param id the ID for the new aggregate
      * @throws IllegalArgumentException if the ID is not of one of the supported types
```
Comment (armiol) [inline common/src/main/java/org/spine3/samples/lobby/common/util/aggregate/AbstractLobbyAggregate.java]: Please explain what the supported types are. Or provide a link to the related functionality (`@see` or `@link`).

File: common/src/main/proto/spine/test/aggregate/commands.proto [hunk 12]
```diff
@@ -21,30 +21,20 @@ syntax = "proto3";
...
-message AddTask {
-    ProjectId project_id = 1;
-    Task task = 2;
-}
-
-message StartProject {
-    ProjectId project_id = 1;
-}
-
+// A command performed to import (inject) events into the aggregate's processing flow.
```
Comment (armiol) [inline common/src/main/proto/spine/test/aggregate/commands.proto]: You don't need `'s`. It is better to rephrase as `... into the processing flow of an aggregate`.

File: registration/src/main/java/org/spine3/samples/lobby/registration/conference/ConferenceProjection.java [hunk 13]
```diff
@@ -196,19 +194,17 @@ private void sendRemoveSeatsRequest(SeatTypeId seatTypeId, int quantity) {
                                                .build();
         final Command command = create(message, newCommandContext());
         commandBus.post(command, new StreamObserver<Response>() {
```
Comment (armiol) [inline registration/src/main/java/org/spine3/samples/lobby/registration/conference/ConferenceProjection.java:198]: It is much better to add a class-level comment and explain the `StreamObserver` aim.

**Comment by armiol** [comment]
@dmdashenkov please address some of my new comments.

**Comment by dmdashenkov** [comment]
@armiol It turns out that Lobby was still at `v2.8` of Gradle, and minimum required for proto plugin version is higher. Now Lobby and Spine both use `v3.0`.

**Comment by dmdashenkov** [comment]
@armiol PTAL

**Review by armiol — APPROVED** [review]

**Comment by armiol** [comment]
@dmdashenkov LGTM.

**Pull request merged**